sklearn/decomposition/tests/test_pca.py

```diff
@@ -683,15 +683,58 @@ def test_svd_solver_auto():
     assert_array_almost_equal(pca.components_, pca_test.components_)
 
 
-@pytest.mark.parametrize('svd_solver', solver_list)
-def test_pca_sparse_input(svd_solver):
+def test_pca_sparse_input_randomized_solver():
+    rng = np.random.RandomState(0)
+    n_samples = 100
+    n_features = 80
+
+    # The randomized method produces larger errors whenever the means of the
+    # matrix are way off the origin
+    X = rng.normal(1000, 20, (n_samples, n_features))
+
+    X_sp = sp.sparse.csr_matrix(X)
+    assert sp.sparse.issparse(X_sp)
+
+    # Compute the complete decomposition on the dense matrix
+    pca = PCA(n_components=3, svd_solver='full', random_state=rng).fit(X)
```

Comment:
is it necessary ? you can fit both `X` and `X_sp` with `svd_solver=randomized`.

Reply:
I guess you're right, we already check the correctness of the randomized solver in `test_singular_values`.